### PR TITLE
[mlir][python] Fix generation of Python bindings for `async` dialect

### PR DESCRIPTION
**Pull request by adk9**
The Python bindings generated for "async" dialect didn't include any of the "async" dialect ops. This PR fixes issues with generation of Python bindings for "async" dialect and adds a test case to use them.

### CODE REVIEW

File: mlir/python/CMakeLists.txt [hunk 1]
```diff
@@ -710,11 +710,10 @@ if(NOT LLVM_ENABLE_IDE)
     COMPONENT mlir-python-sources
   )
 endif()
-
-################################################################################
+# ###############################################################################
 # The fully assembled package of modules.
 # This must come last.
-################################################################################
+# ###############################################################################
```
Comment (joker-eph) [inline mlir/python/CMakeLists.txt:716]: Is this like auto-formatting changes here?

**Review by joker-eph — APPROVED** [review]

File: mlir/python/CMakeLists.txt (hunk 1 above)
Comment (adk9) [reply mlir/python/CMakeLists.txt]: Yeah, looks like it. I reverted the auto-formatting changes.
Thanks for the review!

**Review by makslevental — APPROVED** [review]
Dunno how I missed this at the time (at a certain point I thought I had python examples that ran through async but I guess not) and also this PR. Thank you very much for fixing this.

**Comment by makslevental** [comment]
@adk9 do you need me to merge for you?

**Comment by adk9** [comment]
> @adk9 do you need me to merge for you?

Yes, please. I don't see an option to merge. Thanks!

File: utils/bazel/llvm-project-overlay/mlir/python/BUILD.bazel [hunk 2]
```diff
@@ -331,6 +331,53 @@ filegroup(
...
+
+gentbl_filegroup(
+    name = "AsyncOpsPyGen",
+    tbl_outs = [
+        (
+            [
+                "-gen-python-enum-bindings",
+                "-bind-dialect=async",
+            ],
+            "mlir/dialects/_async_enum_gen.py",
```
[the remark this answers is not in the document]
Comment (ftynse) [reply utils/bazel/llvm-project-overlay/mlir/python/BUILD.bazel:346]: I removed this in d5093aac543dfd24f339f7d43aa02282f7c16fa5. @adk9 please make sure to keep track of your PRs even after they land. In LLVM, reviews may happen after commit and authors are expected to address them.

[the remark this answers is not in the document]
Comment (adk9) [reply utils/bazel/llvm-project-overlay/mlir/python/BUILD.bazel:346]: > I removed this in [d5093aa](https://github.com/llvm/llvm-project/commit/d5093aac543dfd24f339f7d43aa02282f7c16fa5). @adk9 please make sure to keep track of your PRs even after they land. In LLVM, reviews may happen after commit and authors are expected to address them.

Will definitely do. Thanks for the fix.

**Pull request merged**